src/components/Paginate.tsx

```diff
@@ -0,0 +1,200 @@
+import React, { useEffect, useState } from "react";
+
+const getPageCount = (
+  numberOfResults: number,
+  numberOfResultsPerPage: number
+): number => Math.ceil(numberOfResults / numberOfResultsPerPage);
+
+interface paginateProps {
+  children: any; // children: component?
+  limit: number;
+  onPageClick: Function;
```

Comment:
`Function` is too generic and needs more specific types. Maybe something like

```
(e: SomeReactEvent) => void;
```

Reply:
f2d2f35 addresses this by making sure `onPageClick` takes a `page` argument of type `number`, and returns nothing, with `onPageClick: (page: number) => void;`.

Thank you for checking this!